### PR TITLE
added audio driver for macos

### PR DESCRIPTION
**Pull request by psmokotnin**
Audio driver added for macOS  into MU4 scope

### CODE REVIEW

File: framework/ui/internal/uiengine.cpp [hunk 1]
```diff
@@ -93,12 +93,6 @@ void UiEngine::setup(QQmlEngine* e)
     importPaths.append(dir.absolutePath());
     m_engine->setImportPathList(importPaths);
 #endif
-#ifdef Q_OS_MAC
```
Comment (vpereverzev) [inline framework/ui/internal/uiengine.cpp:96]: What was wrong with that code?

Comment (psmokotnin) [reply framework/ui/internal/uiengine.cpp:96]: This code rewrites qml import paths and application can't find Qt's modules, therefore, crashed on startup with the message, that QtQuick module not found.

File: framework/audio/internal/platform/osx/osxaudiodriver.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_OSXAUDIODRIVER_H
+#define MU_AUDIO_OSXAUDIODRIVER_H
+
+#include <memory>
+#include <MacTypes.h>
```
Comment (vpereverzev) [inline framework/audio/internal/platform/osx/osxaudiodriver.h:23]: I'm wondering why we need this include here?

File: framework/ui/internal/uiengine.cpp (hunk 1 above)
Comment (vpereverzev) [reply framework/ui/internal/uiengine.cpp:96]: ha, quite interesting

File: framework/audio/internal/platform/osx/osxaudiodriver.h (hunk 2 above)
Comment (psmokotnin) [reply framework/audio/internal/platform/osx/osxaudiodriver.h:23]: MacTypes defines OSStatus

**Pull request merged**